optuna/pruners/_wilcoxon.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+
+import optuna
+from optuna._experimental import experimental_class
+from optuna.pruners import BasePruner
+from optuna.study._study_direction import StudyDirection
+from optuna.trial import FrozenTrial
+
+
+@experimental_class("3.6.0")
+class WilcoxonPruner(BasePruner):
+    """Pruner based on the Wilcoxon signed-rank test <https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test>`_.
+
+    This pruner performs the Wilcoxon signed-rank test between the current trial and the current best trial,
+    and stops whenever the pruner is sure up to a given p-value that the current trial is worse than the best one.
+
+    This pruner is effective for objective functions (median, mean, etc.) that
+    aggregates multiple evaluations.
+    This includes the mean performance of n (e.g., 100)
+    shuffled inputs, the mean performance of k-fold cross validation, etc.
+    There can be "easy" or "hard" inputs (the pruner handles correspondence of
+    the inputs between different trials),
+    but it is recommended to shuffle the order of inputs once before the optimization.
+
+    When you use this pruner, you must call `Trial.report(value, step)` function for each step (e.g., input id) with
+    the evaluated value. This is different from other pruners in that the reported value need not converge
+    to the real value. (To use pruners such as `SuccessiveHalvingPruner` in the same setting, you must provide e.g.,
+    the historical average of the evaluated values.)
+
+    .. seealso::
+        Please refer to :meth:`~optuna.trial.Trial.report`.
+
+    Example:
+
+        .. testcode::
+
+            import optuna
+            import numpy as np
+
+
+            # For demonstrative purposes, we will use a toy evaluation function.
+            # We will minimize the mean value of `eval_func` over the input dataset.
+            def eval_func(param, input_):
+                return (param - input_) ** 2
+
+
+            input_data = np.linspace(-1, 1, 100)
+
+            # It is recommended to shuffle the input data once before optimization.
+            np.random.shuffle(input_data)
+
+
+            def objective(trial):
+                s = 0.0
+                for i in range(len(input_data)):
+                    param = trial.suggest_uniform("param", -1, 1)
```

Comment:
According to the following warning, please use `suggest_float` instead of `suggest_uniform`.
```
wp.py:20: FutureWarning: suggest_uniform has been deprecated in v3.0.0. This feature will be removed in v6.0.0. See https://github.com/optuna/optuna/releases/tag/v3.0.0. Use suggest_float instead. 
```

Reply:
```suggestion
                    param = trial.suggest_float("param", -1, 1)
```